xarray/tests/test_datatree.py

```diff
@@ -2,46 +2,47 @@
 
 import numpy as np
 import pytest
+
 import xarray as xr
+import xarray.datatree_.datatree.testing as dtt
 import xarray.testing as xrt
+from xarray.core.datatree import DataTree
+from xarray.core.treenode import NotFoundInTreeError
 from xarray.tests import create_test_data, source_ndarray
 
-import xarray.datatree_.datatree.testing as dtt
-from xarray.datatree_.datatree import DataTree, NotFoundInTreeError
-
 
 class TestTreeCreation:
     def test_empty(self):
-        dt = DataTree(name="root")
+        dt: DataTree = DataTree(name="root")
         assert dt.name == "root"
         assert dt.parent is None
         assert dt.children == {}
         xrt.assert_identical(dt.to_dataset(), xr.Dataset())
 
     def test_unnamed(self):
-        dt = DataTree()
+        dt: DataTree = DataTree()
         assert dt.name is None
 
     def test_bad_names(self):
         with pytest.raises(TypeError):
-            DataTree(name=5)
+            DataTree(name=5)  # type: ignore[arg-type]
```

Comment:
I have added a few `mypy` annotations like this. I think they make sense, because the tests are explicitly checking what happens when you use the wrong type of argument.

Reply:
Yes you always have to do this within tests that check for `TypeErrors`

Reply:
Cool beans. I didn't want people to think I'd just thrown in `type: ignore` statements in just to quieten down `mypy`. (I'd thought about it a bit, and _then_ thrown in `type: ignore` statements to quieten down `mypy` :wink:)